### PR TITLE
bpo-35755: Remove current directory from posixpath.defpath

### PR DESCRIPTION
**Pull request by vstinner**
Document the change in a NEWS entry of the Security category.

https://bugs.python.org/issue35755

### CODE REVIEW

File: Lib/test/test_subprocess.py [hunk 1]
```diff
@@ -1378,6 +1378,29 @@ def test_file_not_found_with_bad_cwd(self):
             subprocess.Popen(['exit', '0'], cwd='/some/nonexistent/directory')
         self.assertEqual(c.exception.filename, '/some/nonexistent/directory')
 
+    @support.skip_unless_symlink
+    def test_defpath(self):
+        # bpo-35755: Test that Python doesn't run a program from the current
+        # directory when the PATH environment variable is not test.
```
Comment (asottile) [inline Lib/test/test_subprocess.py]:
```suggestion
        # directory when the PATH environment variable is not set.
```

File: Misc/NEWS.d/next/Security/2019-01-17-10-03-48.bpo-35755.GmllIs.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Remove the current directory from :data:`os.defpath`
+(:data:`posixpath.defpath and :data:`ntpath.defpath`). As a consequence,
```
Comment (asottile) [inline Misc/NEWS.d/next/Security/2019-01-17-10-03-48.bpo-35755.GmllIs.rst]:
```suggestion
(:data:`posixpath.defpath` and :data:`ntpath.defpath`). As a consequence,
```

**Comment by vstinner** [comment]
Hum. The test fails on Windows:

```
FAIL: test_defpath (test.test_subprocess.ProcessTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\cpython\lib\test\test_subprocess.py", line 1402, in test_defpath
    subprocess.run(args, env=env)
AssertionError: FileNotFoundError not raised
```

File: Lib/test/test_subprocess.py [hunk 3]
```diff
@@ -1378,6 +1378,29 @@ def test_file_not_found_with_bad_cwd(self):
...
+        # directory when the PATH environment variable is not set.
+        # Test directly the value of os.defpath.
+        program = 'subprocess_test_defpath.exe'
+        args = (program, '-I', '-S', '-c', 'pass')
+
+        env = dict(os.environ)
+        env.pop('PATH', None)
+
+        with support.temp_cwd() as temp_dir:
+            self.addCleanup(support.rmtree, temp_dir)
```
Comment (vstinner) [inline Lib/test/test_subprocess.py]: Oh. It's now useless, I forgot to remove it. I will remove it once I understand the Windows failure.

**Comment by asottile** [comment]
> 
> 
> Hum. The test fails on Windows:
> 
> ```
> FAIL: test_defpath (test.test_subprocess.ProcessTestCase)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "C:\projects\cpython\lib\test\test_subprocess.py", line 1402, in test_defpath
>     subprocess.run(args, env=env)
> AssertionError: FileNotFoundError not raised
> ```

iirc the behaviour you're seeing is due to https://bugs.python.org/issue8557

**Comment by vstinner** [comment]
I wrote PR #12858 to define the behavior when the PATH environment variable is set to an empty string.

**Pull request merged**